### PR TITLE
add dockerfile

### PR DESCRIPTION
**Pull request by ferrine**
Testing in a reproducible environment become an issue, Docker should solve that problem

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
-  - black --check --diff geoopt tests
-  - py.test -v --durations=0 $COVERAGE
+  - if [[ $COVERAGE ]]; then black --check --diff geoopt tests; fi
+  - ./scripts/container.sh --test
 
 after_success:
-  - coveralls
+  - echo $(pwd)
+  - ls -la
+  - if [[ $COVERAGE ]]; then coveralls; fi
```
Comment (canyon289) [inline .travis.yml]: You may need to modify the coveralls output due to container paths. Here's what we did in ArviZ

https://github.com/arviz-devs/arviz/blob/master/.travis.yml#L73

File: scripts/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM conda/miniconda3
+
+MAINTAINER Maxim Kochurov <maxim.v.kochurov@gmail.com>
```
Comment (canyon289) [inline scripts/Dockerfile]: Maintainer is deprecated. Suggestion for new pattern is in the docs

https://docs.docker.com/engine/reference/builder/#maintainer-deprecated

File: scripts/container.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        --build-arg PYTORCH=${PYTORCH} \
+        --build-arg PYTHON_VERSION=${PYTHON_VERSION} \
+        --rm
+fi
+
+if [[ $* == *--test* ]]; then
+    echo "Testing Geoopt"
+    docker run --rm -it --user $(id -u):$(id -g) --mount type=bind,source="$(pwd)",target=/opt/geoopt/ geoopt:latest \
+        bash -c "pytest --durations=0 -v tests/ ${COVERAGE}" && \
+        if [[ ${COVERAGE} ]]; then sed -i 's@/opt/geoopt@'${SRC_DIR}'@g' "$(pwd)/.coverage"; fi
```
Comment (canyon289) [inline scripts/container.sh]: Ah ignore my comment above, it looks like you do the path changing here

**Comment by canyon289** [comment]
Overall looks good to me from a quick review. Let me know if you want me to dive more in depth

File: scripts/container.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# prototype is taken from  https://github.com/arviz-devs/arviz/blob/master/scripts/container.sh
+SRC_DIR=${SRC_DIR:-`pwd`}
+COVERAGE=${COVERAGE:-"--cov geoopt"}
```
Comment (CoderINusE) [inline scripts/container.sh]: Do we need to set default value for `$COVERAGE` here?

**Comment by ferrine** [comment]
I think yes, usually we want coverage to be collected. You can even see the
report with coverage report

ср, 10 апр. 2019 г., 10:58 Rasul Kerimov <notifications@github.com>:

**Comment by ferrine** [comment]
Any more to say, @CoderINusE, @canyon289?

File: scripts/container.sh (hunk 3 above)
Comment (ferrine) [reply scripts/container.sh]: Yes, I found it works locally as well and hence more convenient

**Pull request merged**